### ISSUE TITLE
SWIFT_VERSION build setting not being applied @1.7.1

### ISSUE THREAD

**Issue by NeoLSN** [akofman/cordova-plugin-add-swift-support#47]
After this issue https://github.com/akofman/cordova-plugin-add-swift-support/issues/43
Now I can restore this plugin without any error message, however, this plugin is not setting swift version.

I'll get this message when I tried to build my app.
```
The “Swift Language Version” (SWIFT_VERSION) build setting must be set to a supported value for targets which use Swift. This setting can be set in the build settings editor.
```

The reproduce step.
1. Create a new cordova project or remove platform folder from project. Make sure there is no platform folder under project.
2. Trying to add iOS platform.
3. Adding a plugin which is using swift, you can try with this - https://github.com/becvert/cordova-plugin-zeroconf.
4. Trying to build an iOS app.

The plugin, `cordova-plugin-zeroconf`, when `cordova-plugin-zeroconf@1.3.0` is using `cordova-plugin-add-swift-support@1.7.0`. It works fine.
When `cordova-plugin-zeroconf@1.3.1` which is using `cordova-plugin-add-swift-support@1.7.1`. I always get the failed message.

**Comment by predorock** [comment]
👍

**Comment by NeoLSN** [comment]
@akofman Can you look at this issue? The plugin hook is not fired on `cordova-plugin-add-swift-support@1.7.1` when adding platform.

**Comment by akofman** [comment]
Hmmmm ... yes ! :) Maybe this evening, I'll keep you in touch guys.

**Comment by akofman** [comment]
I've checked with cordova 8.0 and `cordova-plugin-zeroconf@1.3.1` and it works well:

```
Installing "cordova-plugin-zeroconf" for ios
Installing "cordova-plugin-add-swift-support" for ios
Adding cordova-plugin-zeroconf to package.json
Saved plugin info for "cordova-plugin-zeroconf" to config.xml
Update IOS build setting SWIFT_OBJC_BRIDGING_HEADER to: "$(PROJECT_DIR)/$(PROJECT_NAME)/Bridging-Header.h" for build configuration Debug
Update IOS build setting SWIFT_OBJC_BRIDGING_HEADER to: "$(PROJECT_DIR)/$(PROJECT_NAME)/Bridging-Header.h" for build configuration Release
Importing ZeroConf-Bridging-Header.h into /Users/akofman/Programming/cordova-test/platforms/ios/Test/Bridging-Header.h
Update IOS build setting ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES to: YES for build configuration Debug
Update SWIFT version to 3.0 Debug
Update IOS build setting SWIFT_OPTIMIZATION_LEVEL to: -Onone for build configuration Debug
Update IOS build setting ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES to: YES for build configuration Release
Update SWIFT version to 3.0 Release
```

I'm going to check with a legacy version of cordova, what version are you using @NeoLSN, @predorock ?

**Comment by NeoLSN** [comment]
```
cordova platform add ios --save
Using cordova-fetch for cordova-ios@4.5.4

Adding ios project...

Creating Cordova project for the iOS platform:

        Path: platforms/ios
        Package: <project_package>
        Name: <project_Name>

iOS project created with cordova-ios@4.5.4

Discovered plugin "cordova-open-native-settings" in config.xml. Adding it to the project

Installing "cordova-open-native-settings" for ios

Adding cordova-open-native-settings to package.json

Saved plugin info for "cordova-open-native-settings" to config.xml
Discovered plugin "cordova-plugin-advanced-http" in config.xml. Adding it to the project
Installing "cordova-plugin-advanced-http" for ios

Installing "cordova-plugin-file" for ios

Adding cordova-plugin-advanced-http to package.json
Saved plugin info for "cordova-plugin-advanced-http" to config.xml

Discovered plugin "cordova-plugin-app-version" in config.xml. Adding it to the project

Installing "cordova-plugin-app-version" for ios

Adding cordova-plugin-app-version to package.json

Saved plugin info for "cordova-plugin-app-version" to config.xml

Discovered plugin "cordova-plugin-device" in config.xml. Adding it to the project

Installing "cordova-plugin-device" for ios

Adding cordova-plugin-device to package.json
Saved plugin info for "cordova-plugin-device" to config.xml

Discovered plugin "cordova-plugin-email-composer" in config.xml. Adding it to the project

Installing "cordova-plugin-email-composer" for ios

Adding cordova-plugin-email-composer to package.json

Saved plugin info for "cordova-plugin-email-composer" to config.xml

Discovered plugin "cordova-plugin-facebook4" in config.xml. Adding it to the project
Installing "cordova-plugin-facebook4" for ios

Adding cordova-plugin-facebook4 to package.json

Saved plugin info for "cordova-plugin-facebook4" to config.xml

Discovered plugin "cordova-plugin-hockeyapp" in config.xml. Adding it to the project

Installing "cordova-plugin-hockeyapp" for ios

Adding cordova-plugin-hockeyapp to package.json

Saved plugin info for "cordova-plugin-hockeyapp" to config.xml

Discovered plugin "cordova-plugin-inappbrowser" in config.xml. Adding it to the project
Installing "cordova-plugin-inappbrowser" for ios

Adding cordova-plugin-inappbrowser to package.json

Saved plugin info for "cordova-plugin-inappbrowser" to config.xml

Discovered plugin "cordova-plugin-insomnia" in config.xml. Adding it to the project

Installing "cordova-plugin-insomnia" for ios

Adding cordova-plugin-insomnia to package.json

Saved plugin info for "cordova-plugin-insomnia" to config.xml

Discovered plugin "cordova-plugin-ionic-webview" in config.xml. Adding it to the project

Installing "cordova-plugin-ionic-webview" for ios
Adding cordova-plugin-ionic-webview to package.json

Saved plugin info for "cordova-plugin-ionic-webview" to config.xml
Discovered plugin "cordova-plugin-market" in config.xml. Adding it to the project
Installing "cordova-plugin-market" for ios

Adding cordova-plugin-market to package.json

Saved plugin info for "cordova-plugin-market" to config.xml
Discovered plugin "cordova-plugin-network-information" in config.xml. Adding it to the project

Installing "cordova-plugin-network-information" for ios

Adding cordova-plugin-network-information to package.json

Saved plugin info for "cordova-plugin-network-information" to config.xml
Discovered plugin "cordova-plugin-safariviewcontroller" in config.xml. Adding it to the project
Installing "cordova-plugin-safariviewcontroller" for ios

Adding cordova-plugin-safariviewcontroller to package.json

Saved plugin info for "cordova-plugin-safariviewcontroller" to config.xml

Discovered plugin "cordova-plugin-splashscreen" in config.xml. Adding it to the project

Installing "cordova-plugin-splashscreen" for ios

Adding cordova-plugin-splashscreen to package.json
Saved plugin info for "cordova-plugin-splashscreen" to config.xml
Discovered plugin "cordova-plugin-whitelist" in config.xml. Adding it to the project

Installing "cordova-plugin-whitelist" for ios

Adding cordova-plugin-whitelist to package.json

Saved plugin info for "cordova-plugin-whitelist" to config.xml

Discovered plugin "cordova-plugin-x-socialsharing" in config.xml. Adding it to the project

Installing "cordova-plugin-x-socialsharing" for ios

Installing "es6-promise-plugin" for ios

Adding cordova-plugin-x-socialsharing to package.json

Saved plugin info for "cordova-plugin-x-socialsharing" to config.xml

Discovered plugin "cordova-sqlite-storage" in config.xml. Adding it to the project

Installing "cordova-sqlite-storage" for ios

installing external dependencies via npm
npm install of external dependencies ok
Adding cordova-sqlite-storage to package.json

Saved plugin info for "cordova-sqlite-storage" to config.xml

Discovered plugin "cordova-plugin-zeroconf" in config.xml. Adding it to the project

Installing "cordova-plugin-zeroconf" for ios

Installing "cordova-plugin-add-swift-support" for ios

Adding cordova-plugin-zeroconf to package.json

Saved plugin info for "cordova-plugin-zeroconf" to config.xml
Discovered plugin "ionic-plugin-keyboard" in config.xml. Adding it to the project

Installing "ionic-plugin-keyboard" for ios

Adding ionic-plugin-keyboard to package.json

Saved plugin info for "ionic-plugin-keyboard" to config.xml

Discovered plugin "ionic-plugin-deeplinks" in config.xml. Adding it to the project

Installing "ionic-plugin-deeplinks" for ios

Installing dependency packages:
{
  "mkpath": ">=1.0.0",
  "node-version-compare": ">=1.0.1",
  "plist": ">=1.2.0",
  "xml2js": ">=0.4"
}
Adding ionic-plugin-deeplinks to package.json

Saved plugin info for "ionic-plugin-deeplinks" to config.xml

IOS project now has deployment target set as: 8.0
IOS project Code Sign Entitlements now set to: <project>.entitlements
Entitlements file is not in references section, adding it
--save flag or autosave detected

Saving ios@~4.5.4 into config.xml file ...
```

I do the restore from a git cloned project. The hook is never fired.

**Comment by akofman** [comment]
I'm not familiar with ionic but if it doesn't work since the `1.7.1` version, I guess the cause of this issue is somewhere here: https://github.com/akofman/cordova-plugin-add-swift-support/blob/master/src/main.js#L199

I suspect the platform version not to be found. Are you able to check the Cordova project root in your ionic project ? This plugin tries to get the version from the one indicated here: `CORDOVA_PROJECT_ROOT/platforms/ios/cordova/version`, do you have this file in your project ?

**Comment by NeoLSN** [comment]
Yes, I have.
![image](https://user-images.githubusercontent.com/5782106/37206458-a7b5d1ea-23d3-11e8-8e00-56fb98953b03.png)
![image](https://user-images.githubusercontent.com/5782106/37206507-cec5193a-23d3-11e8-9b8a-5084679a41c4.png)

**Comment by akofman** [comment]
hmmmm, okay I have to test with an ionic project ... Did you try to remove all your plugins except the one with the dependency to `cordova-plugin-add-swift-support` ? Or maybe you can try to add it directly to your project. Sorry for the wait, I will try asap with an ionic sample.

**Comment by NeoLSN** [comment]
Yes, I have tried those cases before I reported this bug.
1. Restore a git cloned project - failed.
2. Remove plugin then add it back - failed.
3. Create a new project and add this plugin - failed.

**Comment by NeoLSN** [comment]
@akofman Is there any new progress?

**Comment by TheLiquidSky** [comment]
We had the exact same error message when we tried to build our app. As @akofman suggested the platform version could indeed not be detected but  in our case that was due to our project being located in a location that had white spaces (and brackets) in it's path. I have supplied a fix in PR #52 with more technical details.

**Comment by akofman** [comment]
@NeoLSN please confirm it to us but I think PR #52 is fixing this issue. I'm releasing a 1.7.2 version.
